### PR TITLE
bump go to 1.16

### PR DESCRIPTION
**Pull request by guangbochen**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
Bump Harvester go dependency to 1.16 to make it consistent with Rancher 2.6

**Solution:**
Bump Harvester Go dependency to 1.16

**Related Issue:**
https://github.com/harvester/harvester/issues/916

**Test plan:**
N/A

### CODE REVIEW

**Comment by bk201** [comment]
@gitlawr suggests fixing the kubevirt API doc dependency with a blank import:

```diff
diff --git a/pkg/genswagger/main.go b/pkg/genswagger/main.go
index 5eb2518f..02f4c5bf 100644
--- a/pkg/genswagger/main.go
+++ b/pkg/genswagger/main.go
@@ -11,6 +11,7 @@ import (
        "github.com/go-openapi/spec"
        "k8s.io/kube-openapi/pkg/builder"
        "k8s.io/kube-openapi/pkg/common"
+       _ "kubevirt.io/client-go/apis/snapshot/v1alpha1"

        "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
        "github.com/harvester/harvester/pkg/genswagger/rest"
```

File: .golangci.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
 		"disable-all": true,
 		"enable": [
 			"govet",
-			"golint",
```
Comment (gitlawr) [inline .golangci.json:6]: Why remove this

Comment (guangbochen) [reply .golangci.json:6]: this repo is archived and using govet is sufficient enough. https://github.com/golang/lint

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**